extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/ContractAgreementTerminationApiService.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2024 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.ext.wrapper.api.ui.pages.contract_agreements;
+
+import de.sovity.edc.ext.wrapper.api.ui.model.ContractTerminationRequest;
+import de.sovity.edc.ext.wrapper.api.ui.model.IdResponseDto;
+import de.sovity.edc.extension.contacttermination.ContractAgreementTerminationService;
+import de.sovity.edc.extension.contacttermination.ContractTermination;
+import jakarta.validation.Validation;
+import jakarta.ws.rs.BadRequestException;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+
+@RequiredArgsConstructor
+public class ContractAgreementTerminationApiService {
+
+    private final ContractAgreementTerminationService contractAgreementTerminationService;
+
+    public IdResponseDto terminate(
+        String contractAgreementId,
+        ContractTerminationRequest contractTerminationRequest) {
+
+        try(val f = Validation.buildDefaultValidatorFactory()){
+            val validator = f.getValidator();
+            val constraintViolations = validator.validate(contractTerminationRequest);
+            if (!constraintViolations.isEmpty()) {
+                throw new BadRequestException("Invalid contract termination request: " + constraintViolations);
```

Comment:
comment: You could also throw a ConstraintViolationException which is what the validator would throw normally. But feel free to ignore this if we're going to implement the automatic validation anyway in the future

Reply:
Idea: pitch SO that we could enable this in the EDC for everyone if it doesn't exist.

Reply:
Also, this is now changed to a `ConstraintViolationException`